src/groupeddataframe/complextransforms.jl

```diff
@@ -30,28 +30,30 @@ function _combine_with_first((first,)::Ref{Any},
     @assert first isa Union{NamedTuple, DataFrameRow, AbstractDataFrame}
     extrude = false
 
+    lgd = length(gd)
     if first isa AbstractDataFrame
         n = 0
         eltys = eltype.(eachcol(first))
     elseif first isa NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}}
         n = 0
         eltys = map(eltype, first)
     elseif first isa DataFrameRow
-        n = length(gd)
+        n = lgd
         eltys = [eltype(parent(first)[!, i]) for i in parentcols(index(first))]
     elseif !firstmulticol && first[1] isa Union{AbstractArray{<:Any, 0}, Ref}
         extrude = true
         first = wrap_row(first[1], firstcoltype(firstmulticol))
-        n = length(gd)
+        n = lgd
         eltys = (typeof(first[1]),)
     else # other NamedTuple giving a single row
-        n = length(gd)
+        n = lgd
         eltys = map(typeof, first)
         if any(x -> x <: AbstractVector, eltys)
             throw(ArgumentError("mixing single values and vectors in a named tuple is not allowed"))
         end
     end
     idx = idx_agg === NOTHING_IDX_AGG ? Vector{Int}(undef, n) : idx_agg
+    sizehint!(idx, lgd)
```

Comment:
This assumes that there's going to be at least one row per group, right? Maybe add a comment explaining this? That's not necessarily the case if groups are dropped, though that's probably quite common.

Reply:
Added. What you write is exactly the thinking. `sizehint!` is cheap and dropping groups is uncommon (actually in the past it was even super hard to do :smile:):
```
julia> x = Int[];

julia> @time sizehint!(x, 10^7);
  0.000018 seconds (2 allocations: 76.294 MiB)

julia> x = Int[];

julia> @time sizehint!(x, 10^7);
  0.000021 seconds (2 allocations: 76.294 MiB)
```